# Lab 01 - Introduction to Experimentation

Today's notebook has some reminders and template 'scaffolding' in it to help you make
sure you put all the pieces in it. As we go through the next few weeks, you'll see less and
less of that.

Please start by **running the cell below** (by clicking on it and pressing the play/triangle button, or by Shift + Return) before getting started! This cell will clear the python environment and import the necessary Python packages to run your data analysis in today's lab.

In [8]:
# This clears the python workspace of any previously defined variables.:
%reset -f 

# Import the numpy library:
import numpy as np

Edit this cell and provide the information requested:

- Your name: Mary
- Your student number: 
- Partner's name: Musa


## Part B - Familiarization with the experiment

Double click on this cell to edit it! Write down any notes you have from the familiarization here. Consult the lab instructions for tips on what to include in your lab notes during this familiarization stage. Remember that you can use bullet points if you wish:
- Uncompressed mass is 0.078kg
- Height of spring when uncompressed 13cm
- Compressed mass is 0.700
- Height of spring when compressed 7.6cm

#### How did you take your measurement?

First we measured the height of the spring. Since the spring was slanted on top, we took the average of the slant by measuring the highest, middle and lowest point of the slant. Then we took the average of our measurements (12.9, 13 and 13.1) and got 13cm as the average total height of the spring.

Then we placed the spring on the scale to get the uncompressed mass of 0.078kg. 

After which I placed my hand and compressed the spring while my partner measured the height of the spring. We recorded a compressed mass of 0.700kg as the mass and a compressed height of 7.6cm.

**Side Note:** When measuring the height of the spring, it is important to mention that we measured from the table and subtract the height of the scale from the measurement that we got. So the finalized measurements that we reported is the result after we subtract the height of the scale. 



## Your initial measurement and calculation of $k$

Use the Python cell below to enter your measured values for $m_1$, $m_2$, $x_1$, and $x_2$. You may then calculate $k$ from these Python variables within the same cell.

In [9]:
# type in your measured values to the right of the '=' to define each Python variable below
m1 = 0.078 # uncompressed mass reading
x1 = 0.13   # uncompressed height
m2 = 0.700  # compressed mass reading
x2 = 0.076  # compressed height
g = 9.809   # m/s^2
# write some code below to calculate k
dm = np.abs(m2 - m1)
dx = np.abs(x2 - x1)
k = (dm/dx)*g
k
# it may be helpful to define intermediate variables, e.g. delta_m, to make your calculation easier to follow

112.98514814814813

# Part C - Discussion with your group (Familiarization)

Write down notes from comparisons/discussions with your group here. Tips below:
- How do your methods and results compare to other members of your group? The other two groups had higher k value (160 and 180) whereas ours is lower (113). We suspect this might be due to the amount of compression of the spring between use and the other two groups. We also took the average of the slant whereas they recorded the highest point of the spring as its height.
- Did they experience challenges or insights different from yours? We all had a similar issues/difficulties. For example we had issues on how to hold the spring when compressing it. The other groups also had some issues with unit changes for the calculations (forgot to convert cm to m) whereas we forgot to double check our measurements (added an extra 0).

# Part D -  How the instructional team would perform the measurement

Follow along with the slides and make some notes as you go. Make sure to also write down the most important equations and a brief note on how/when to use it (for future reference). Consult the appendix in the Lab 1 instructions document on Canvas for Markdown code for the equations, which you can copy into this cell.

Below is an example of a Markdown formatted equation with notes (which you should remove and replace with your own notes).

**Quadratic formula:**

$$ x_{root} = \frac{ -b \pm \sqrt{b^2 - 4ac} }{ 2a } $$

- The quadratic formula provides a solution for the roots (or zeros) of a quadratic equation $ax^2 + bx + c = 0$
- The discriminant, $ b^2 - 4ac $, tells us how many roots the equation has
    - $ b^2 - 4ac < 0 $ means no roots
    - $ b^2 - 4ac = 0 $ means one root
    - $ b^2 - 4ac > 0 $ means two roots

#### New tools that you learned (there should be 4):

- Mean: $$\bar{x} = \frac{1}{N}\sum_{i=1}^N x_i $$
    - Basically the average of something 
- Standard Deviation Formula: $$ \sigma = \sqrt{\frac{1}{N-1} \sum_{i=1}^N (x_i - \bar{x})^2} $$
    - Measures how wide the distribution is
    - Think of the average distance between each point example
- Rectangular Probability Density Function: $$\sigma_{rectangular}= \frac{a}{\sqrt{3}}$$
    - Straightforward, no measuring involved, single numerical value
- Gaussian Probability Density Function: $$\sigma_{Gaussian} = (95\% CI)/4$$
    - Will have variability with the measurement (uncertainty, eg. human fluctuation)
- Uncertainty: 
$$|\Delta m| = |m_2 - m_1|$$
$$u_{\Delta m} = \sqrt{(u\_m_1)^2+(u\_m_2)^2}$$
    - Estimate of uncertainty in the difference

#### Answers to the uncertainty example discussion

1. Gaussian: the measurements for a ball dropping will be different every time (human fluctuation). Therefore there will different results resulting in Gaussian PDF.
2. Rectangular: the thermometer will output one result. We wouldn't have to measure anything as the thermometer will do the measuring and output a single number. Therefore, rectangular PDF.
3. Gaussian: There will be a similar issue as the one mentioned in 1, human fluctuation. Each person can record a different pressure causing variability. Therefore Gaussian PDF is more appropriate.

# Part E - Revisiting your spring constant measurement with measurement uncertainty characterizations and calculations 

## Notes from the investigation stage

This is the main ***investigation*** after your ***familiarization***. Explain how you do your measurements in enough detail that someone else could repeat it. Tips below:
- What type of uncertainty does each variable have? Why? How did you calculate uncertainty for each variable? Justify your uncertainty choices for each measured quantity.
    - The compressed height is Gaussian PDF
    - The compressed mass is Gaussian PDF
    - as well as the uncompressed height is Gaussian PDF
    - The uncompressed mass has is rectangular PDF
- Read the lab instructions and questions carefully.
- Check the grading rubric for further information on what we are looking for in this week’s grading.

## Measurement and data analysis

Add a new Python cell below and adapt your code from Part A to perform your improved measurement of $k$, along with uncertainty calculations. Consult the Appendix in the student instructions for helpful Python code which you can adapt to perform your uncertainty propagation calculations.

In [10]:
g = 9.809

x1_max = 0.1
u_x1 = (x1_max*0.95)/4

x2_max = 0.077
x2_min = 0.075
u_x2 = (x2_max - x2_min)/4

u_x = np.sqrt(u_x1**2 + u_x2**2)


u_m1 = 0.5/np.sqrt(3)

m2_max = 0.743
m2_min = 0.700
u_m2 = (m2_max - m2_min)/4

u_m = np.sqrt(u_m1**2 + u_m2**2)
print(u_m, u_x)

u_deltaX = u_m * g
u_deltaF = u_x * g

deltaX = (u_m2-u_m1) * g
deltaF = (u_x2-u_x1) * g

k = (u_m/u_x) * g

#what we want is K + uncertainty of K
u_k = k*np.sqrt((u_deltaF/deltaF)**2 + (u_deltaX/deltaX)**2)
u_k

0.2888752253713242 0.023755262574848546


173.8528362800166

# Part F - Summary of results (Start this by 2h30 after the start of the lab)

You do not need to write a formal conclusion, but tips on what to include are provided below.

- Restate your final result for k, with uncertainty
- Summarize the results of any comparisons you made with others regarding your final result. Are you able to draw any conclusions from these comparisons? (*Hint: the experimental goal may be helpful here*)
- Discuss in 1-2 sentences what you would do next to improve your results if you had additional time to work on this lab

The original value of K we got was 112.9 and the new K value with the uncertainty included is 173.8. The results of the calculations vs the results that we previously got are somewhat similar. The result we got with the uncertainty seems to be more accurate as it is closer to the range of results our classmates got compared to the K value we got originally. This displays that with the uncertainty included in the K value, we are more likely to get an accurate value for K. 

In the next lab, it is better if we continue discussing our methods with our classmates to avoid complicating the procedures. Another way that our results would be improved is if we took a longer time to better understand the steps to get the results that we wanted. We were somewhat scattered with our steps which prolonged our time when calculating. 

# Submit your lab by 2h50m after the start of lab

Steps for submission:

1. Click: Run => Run_All_Cells
2. Read through the notebook to ensure all the cells executed correctly and without error.
3. Correct any errors you find!
4. File => Save_and_Export_Notebook_As->HTML
5. Upload the HTML document to the lab submission assignment on Canvas.

You will receive a participation grade this week for a reasonable submission as well as feedback from a TA for next week.